test/e2e/lib/flows/login-flow.js

```diff
@@ -260,6 +260,12 @@ export default class LoginFlow {
 
 	async loginUsingPopup() {
 		await driverHelper.waitForNumberOfWindows( this.driver, 2 );
+
+		// waitForNumberOfWindows returns after the new window has been created but
+		// before the window is fully loaded. Switching to the new window is possible
+		// but our switchToWindowByIndex attempts a window resize which sometimes fails.
+		// A short sleep here is enough to alleviate this problem.
+		await this.driver.sleep( 100 );
```

Comment:
I'm not sure what is failing here. Could you link to a job where this helper has failed, please?

Reply:
No job I'm aware of, happens relatively frequently when testing  locally. This happened on my second run:

```
e2e trunk NODE_CONFIG_ENV=decrypted BROWSERSIZE=desktop ./node_modules/.bin/mocha specs/wp-likes-spec.js


  [WPCOM] Likes: (desktop)
    Like posts and comments @parallel
Logging in as e2eflowtestinggutenbergsimple
      ✓ Login, create a new post and view it (38692ms)
      ✓ Like post (2442ms)
      ✓ Unlike post (128ms)
      ✓ Post comment (3611ms)
      ✓ Like comment (1233ms)
      ✓ Unlike comment (86ms)
      1) Like post as logged out user
FAILED: Taking screenshot of: 'https://wordpress.co\m/log-in?...snip


  6 passing (58s)
  1 failing

  1) [WPCOM] Likes: (desktop)
       Like posts and comments @parallel
         Like post as logged out user:
     WebDriverError: unknown error: cannot determine loading status
from no such window
  (Session info: chrome=90.0.4430.72)
      at Object.throwDecodedError (/home/louis/wp/wp-calypso/node_modules/selenium-webdriver/lib/error.js:550:15)
      at parseHttpResponse (/home/louis/wp/wp-calypso/node_modules/selenium-webdriver/lib/http.js:565:13)
      at Executor.execute (/home/louis/wp/wp-calypso/node_modules/selenium-webdriver/lib/http.js:491:26)
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
      at async Driver.execute (/home/louis/wp/wp-calypso/node_modules/selenium-webdriver/lib/webdriver.js:700:17)
      at async Window.setRect (/home/louis/wp/wp-calypso/node_modules/selenium-webdriver/lib/webdriver.js:1483:14)
      at async Proxy.resizeBrowser (lib/driver-manager.js:232:5)
      at async Proxy.switchToWindowByIndex (lib/driver-helper.js:371:2)
      at async LoginFlow.loginUsingPopup (lib/flows/login-flow.js:263:3)
      at async Context.<anonymous> (specs/wp-likes-spec.js:95:4)
```

When this failure occurs I can see the login page visible, the username/password doesn't get entered.

Its possible this is a chrome/driver issue but I've attempted to make sure they match.

Reply:
Also suspect we might be able to drop this: https://github.com/Automattic/wp-calypso/blob/e51b2a4a61d05e526e951440a9a3f6c86410bb5f/test/e2e/lib/flows/login-flow.js#L276 but I'm not sure, we might need both.

Reply:
Thanks for all the details! 🙇 The `cannot determine loading status` error is thrown by the `waitUntilLocatedAndVisible` helper in my testing. It happens when it tries to `findElement` in a window that's not fully loaded yet, IIUC. To handle that, [I've wrapped](https://github.com/Automattic/wp-calypso/pull/52090/commits/6fb893ea5efad5d6886257298604eeeca35459f9) the locator with the `try/catch` block so it retries when bumped into **any** exception. Also, to simplify and make the window switching more robust, I've added the `waitUntilAbleToSwitchToWindow` helper that will switch to a window with a given index only when it's truly possible.
 - no `sleep` needed! 💪 Let me know what you think!

Reply:
> Also suspect we might be able to drop this:
> 
> https://github.com/Automattic/wp-calypso/blob/e51b2a4a61d05e526e951440a9a3f6c86410bb5f/test/e2e/lib/flows/login-flow.js#L276
> 
> but I'm not sure, we might need both.

In my testing, this is still needed as it's related to the window suddenly closing and detaching elements as a consequence. In this PR we're trying to handle window opening/loading, which doesn't seem to affect this fallback.

Reply:
>  I've added the waitUntilAbleToSwitchToWindow helper that will switch to a window with a given index only when it's truly possible - no sleep needed! muscle Let me know what you think!

Perfect!